src/material/schematics/ng-generate/mdc-migration/rules/components/list/list-styles.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ClassNameChange, StyleMigrator} from '../../style-migrator';
+
+export class ListStylesMigrator extends StyleMigrator {
+  component = 'list';
+
+  deprecatedPrefix = 'mat-list';
+
+  mixinChanges = [
+    {
+      old: 'list-theme',
+      new: ['mdc-list-theme', 'mdc-list-typography'],
+    },
+  ];
+
+  classChanges: ClassNameChange[] = [
+    {old: `.mat-list-base`, new: `.mat-mdc-list-base`},
+    {old: `.mat-list`, new: `.mat-mdc-list`},
+    {old: `.mat-list-avatar`, new: `.mat-mdc-list-avatar`},
+    {old: `.mat-list-icon`, new: `.mat-mdc-list-icon`},
```

Comment:
The new MDC list has a couple of different classes (more clear ones for e.g. avatar and icon).

The list icon for example now is: `mat-mdc-list-item-icon` (note the `<..>-item-<..>`), same for avatar.